pkg/controller/worker/genericactuator/actuator_reconcile.go

```diff
@@ -212,9 +292,15 @@ func (a *genericActuator) deployMachineDeployments(ctx context.Context, cluster
 		case !clusterAutoscalerUsed:
 			replicas = deployment.Minimum
 		// If the machine deployment does not yet exist we set replicas to min so that the cluster
-		// autoscaler can scale them as required.
+		// autoscaler can scale them as required. But if the machine deployment has some state records
+		// it means that we aither have controll plane migration or someone  intentionally has deleted
```

Comment:
```suggestion
		// it means that we either have control plane migration or someone intentionally has deleted
```

Reply:
I forgot my head home when I wrote this comments